File: Node/TypeORM_ORM/src/bll/services/productService.ts
import { Products } from 'dal/entities/Products';
import { ProductRepository } from '../../dal/repositories/productRepository';

export class ProductService {
  private productRepository = new ProductRepository();
  async getAll() {
    return await this.productRepository.getAll();
  }

  async getById(id: number) {
    return await this.productRepository.getById(id);
  }

  async create(data: Partial<Products>) {
    return await this.productRepository.create(data);
  }

  async update(id: number, data: Partial<Products>) {
    return await this.productRepository.update(id, data);
  }

  async delete(id: number) {
    return await this.productRepository.delete(id);
  }
}

File: Go/SQLBoiler_ORM/bll/services/productService.go
package services

import (
	"SQLBOILER_ORM/dal/entities"
	"SQLBOILER_ORM/dal/repositories"
	"context"
)

type ProductService struct {
	repo *repositories.ProductRepository
}

func NewProductService(repo *repositories.ProductRepository) *ProductService {
	return &ProductService{repo: repo}
}

func (s *ProductService) GetByID(ctx context.Context, id int) (*entities.Product, error) {
	product, err := s.repo.GetByID(ctx, id)
	if err != nil {
		return nil, err
	}
	return product, nil
}

func (s *ProductService) GetAll(ctx context.Context) ([]*entities.Product, error) {
	return s.repo.GetAll(ctx)
}

func (s *ProductService) Create(ctx context.Context, product *entities.Product) (*entities.Product, error) {
	return s.repo.Create(ctx, product)
}

func (s *ProductService) Update(ctx context.Context, product *entities.Product) error {
	return s.repo.Update(ctx, product)
}

func (s *ProductService) Delete(ctx context.Context, id int) error {
	return s.repo.Delete(ctx, id)
}
